### ISSUE TITLE
Do not add all files to EXTERNAL FILES

### ISSUE THREAD

**Issue by zrzka** [omz/Pythonista-Issues#517]
**System Information**

* Pythonista 3.2 (320000), Default interpreter 3.6.1
* iOS 11.2.2, model iPad7,2, resolution (portrait) 2048.0 x 2732.0 @ 2.0

**Description**

Pythonista is constantly adding every file I do open with dock icon long press gesture to the _EXTERNAL FILES_ section. That's fine, but I think that the logic should be enhanced little bit.

* Reveal Pythonista library
* Tap on _EXTERNAL FILES - Open..._
* Open some folder (`blackmamba` from Working Copy in my case)
* Create any file in this folder (`blackmamba/a.txt` for example)
* Reveal dock, long press and tap on `a.txt`
* File is opened (or tab selected)
* But the `a.txt` now sits in the _EXTERNAL FILES_ sections too

This keeps my _EXTERNAL FILES_ section polluted and messy.

Path to this file equals when I open same file directly from the _EXTERNAL FILES_ section - `blackmamba` folder. So, something like this would be nice:

```python
def should_add_to_external_files_section(path):
    for location in external_files_locations:
        if path.startswith(location):
            return False
    return True
```

In other words, if there's a folder in _EXTERNAL FILES_ containing this file (in any level), do not add this file to the _EXTERNAL FILES_, otherwise do.
